apps/web/src/server/actions/organization.ts

```diff
@@ -34,36 +76,311 @@ export async function getOrganizationProjects(id: Organization["id"]) {
   return projects;
 }
 
-export async function getOrganizations() {
+export async function getOrganizations(): Promise<
+  | [
+      {
+        orgId: number | null;
+        orgName: string | null;
+      }[],
+      null,
+    ]
+  | [null, Error]
+> {
   const session = await getServerAuthSession();
 
   if (!session) {
-    throw new Error("You must be signed in to perform this action");
+    return [null, new Error("You must be signed in to perform this action")];
+  }
+
+  return [
+    await db
+      .select({
+        orgId: organizations.id,
+        orgName: organizations.name,
+      })
+      .from(usersToOrganizations)
+      .leftJoin(
+        organizations,
+        eq(usersToOrganizations.organizationId, organizations.id),
+      )
+      .where(eq(usersToOrganizations.memberId, session.user.id)),
+    null,
+  ];
+}
+
+export async function createOrganization({ name }: InsertOrganization) {
```

Comment:
If we cannot return error as `Error` maybe we could change our convention from returning arrays to something like:
```typescript
type OptionReulst = {
  value: unknown;
} | {
  error: string;
}
```

Reply:
I'll work on it in a separate issue